src/org/labkey/test/components/ui/domainproperties/EntityTypeDesigner.java

```diff
@@ -156,6 +190,52 @@ public T setDescription(String description)
         return elementCache().optionalWarningAlert();
     }
 
+    /**
+     * Dialog that allows the user to set the genId value.
+     */
+    public static class GenIdDialog extends ModalDialog
+    {
+
+        public GenIdDialog(WebDriver driver)
+        {
+            this(new ModalDialogFinder(driver).withTitle("Are you sure you want to update genId"));
+        }
+
+        private GenIdDialog(ModalDialogFinder finder)
+        {
+            super(finder);
+        }
+
+        public String getGenId()
+        {
+            return getWrapper().getFormElement(elementCache().input);
+        }
+
+        public GenIdDialog setGenId(String value)
+        {
+            getWrapper().setFormElement(elementCache().input, value);
+            return this;
+        }
+
+        @Override
+        protected ElementCache newElementCache()
+        {
+            return new ElementCache();
+        }
+
+        @Override
+        protected ElementCache elementCache()
+        {
+            return (ElementCache) super.elementCache();
+        }
+
+        protected class ElementCache extends ModalDialog.ElementCache
+        {
+            protected final WebElement input = Locator.input("newgenidval").findWhenNeeded(this);
+        }
+
+    }
```

Comment:
Could make this class non-static and remove the element cache to simplify it greatly:
```
public class GenIdDialog extends ModalDialog
{
    protected final WebElement input = Locator.input("newgenidval").findWhenNeeded(this);

    public GenIdDialog()
    {
        super(new ModalDialogFinder(EntityTypeDesigner.this.getDriver()).withTitle("Are you sure you want to update genId"));
    }

    public String getGenId()
    {
        return getWrapper().getFormElement(input);
    }

    public GenIdDialog setGenId(String value)
    {
        getWrapper().setFormElement(input, value);
        return this;
    }
}
```

Reply:
Nice! Updated.